### PR TITLE
[BUG] fix `HierarchyEnsembleForecaster` returned unexpected predictions if data had only one hierarchy level and forecasters specified by node

### LINKED ISSUE #5489: [BUG] `HierarchyEnsembleForecaster` returns unexpected predictions if data has only one hierarchy level and forecasters specified by node
**Describe the bug**

The `HierarchyEnsembleForecaster` returns unexpected predictions if the hierarchy has only one level and different forecasters are specified per node (`by=node`).

**To Reproduce**
The code snippet below is similar to the examples in the [documentation](https://www.sktime.net/en/latest/api_reference/auto_generated/sktime.forecasting.compose.HierarchyEnsembleForecaster.html) but with only one level instead of two and only 3 bottom nodes instead of 7.

```python
from sktime.forecasting.compose import HierarchyEnsembleForecaster
from sktime.forecasting.naive import NaiveForecaster
from sktime.forecasting.trend import PolynomialTrendForecaster, TrendForecaster
from sktime.utils._testing.hierarchical import _bottom_hier_datagen
y = _bottom_hier_datagen(
        no_bottom_nodes=3,
        no_levels=1,
        random_seed=123
)
y.groupby(["l1_agg"]).sum().index

# Example of by = 'node'
forecasters = [
    ('trend', TrendForecaster(), [("__total")]),
    ('poly', PolynomialTrendForecaster(degree=2), [('l1_node01')]),
]
forecaster = HierarchyEnsembleForecaster(
                forecasters=forecasters,
                by='node', default=NaiveForecaster()
)
forecaster.fit(y, fh=[1, 2, 3])
y_pred = forecaster.predict()
y_pred
```
*Output:*
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th></th>
      <th>passengers</th>
    </tr>
    <tr>
      <th>l1_agg</th>
      <th>timepoints</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th rowspan="3" valign="top">__total</th>
      <th>1961-01</th>
      <td>9024.895980</td>
    </tr>
    <tr>
      <th>1961-02</th>
      <td>9075.494395</td>
    </tr>
    <tr>
      <th>1961-03</th>
      <td>9126.092810</td>
    </tr>
    <tr>
      <th rowspan="6" valign="top">l1_node01</th>
      <th>1961-01</th>
      <td>497.671704</td>
    </tr>
    <tr>
      <th>1961-01</th>
      <td>432.000000</td>
    </tr>
    <tr>
      <th>1961-02</th>
      <td>501.352085</td>
    </tr>
    <tr>
      <th>1961-02</th>
      <td>432.000000</td>
    </tr>
    <tr>
      <th>1961-03</th>
      <td>505.046482</td>
    </tr>
    <tr>
      <th>1961-03</th>
      <td>432.000000</td>
    </tr>
    <tr>
      <th rowspan="3" valign="top">l1_node02</th>
      <th>1961-01</th>
      <td>5893.819230</td>
    </tr>
    <tr>
      <th>1961-02</th>
      <td>5893.819230</td>
    </tr>
    <tr>
      <th>1961-03</th>
      <td>5893.819230</td>
    </tr>
    <tr>
      <th rowspan="3" valign="top">l1_node03</th>
      <th>1961-01</th>
      <td>1919.395692</td>
    </tr>
    <tr>
      <th>1961-02</th>
      <td>1919.395692</td>
    </tr>
    <tr>
      <th>1961-03</th>
      <td>1919.395692</td>
    </tr>
  </tbody>
</table>
</div>

**Expected behavior**
*Output:*
<div>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th></th>
      <th>passengers</th>
    </tr>
    <tr>
      <th>l1_agg</th>
      <th>timepoints</th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th rowspan="3" valign="top">__total</th>
      <th>1961-01</th>
      <td>9024.895980</td>
    </tr>
    <tr>
      <th>1961-02</th>
      <td>9075.494395</td>
    </tr>
    <tr>
      <th>1961-03</th>
      <td>9126.092810</td>
    </tr>
    <tr>
      <th rowspan="3" valign="top">l1_node01</th>
      <th>1961-01</th>
      <td>497.671704</td>
    </tr>
    <tr>
      <th>1961-02</th>
      <td>501.352085</td>
    </tr>
    <tr>
      <th>1961-03</th>
      <td>505.046482</td>
    </tr>
    <tr>
      <th rowspan="3" valign="top">l1_node02</th>
      <th>1961-01</th>
      <td>5893.819230</td>
    </tr>
    <tr>
      <th>1961-02</th>
      <td>5893.819230</td>
    </tr>
    <tr>
      <th>1961-03</th>
      <td>5893.819230</td>
    </tr>
    <tr>
      <th rowspan="3" valign="top">l1_node03</th>
      <th>1961-01</th>
      <td>1919.395692</td>
    </tr>
    <tr>
      <th>1961-02</th>
      <td>1919.395692</td>
    </tr>
    <tr>
      <th>1961-03</th>
      <td>1919.395692</td>
    </tr>
  </tbody>
</table>
</div>

**Additional context**
The predictions for the node `l1_node01` are duplicated. The first occurrences are the predictions from `PolynomialTrendForecaster`. The second occurrences are predictions of the `NaiveForecaster` which is configured as default. If using two hierarchy levels as in the documentation the predictions are as expected (i.e., no duplicates from the default predictor).

I tried to stay close to the example in the doc. However, maybe I am just using the `HierarchyEnsembleForecaster` class wrong.

**Versions**

System:
    python: 3.11.5 | packaged by conda-forge | (main, Aug 27 2023, 03:35:23) [Clang 15.0.7 ]
executable: ../anaconda3/envs/sktime311/bin/python
   machine: macOS-14.0-x86_64-i386-64bit

Python dependencies:
          pip: 23.2.1
       sktime: 0.24.0
      sklearn: 1.3.0
       skbase: 0.5.1
        numpy: 1.25.2
        scipy: 1.11.2
       pandas: 2.0.3
   matplotlib: 3.7.2
       joblib: 1.3.2
        numba: None
  statsmodels: 0.14.0
     pmdarima: 2.0.3
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
tensorflow_probability: None

### PR DESCRIPTION
**Pull request by VyomkeshVyas**
#### Reference Issues/PRs

Fixes #5489

#### What does this implement/fix? Explain your changes.

This PR fixes bug reported for HierarchyEnsembleForecaster which returned unexpected predictions if data had only one hierarchy level and forecasters specified by node.

#### Does your contribution introduce a new dependency? If yes, which one?

No

#### Did you add any tests for the change?

Not added, but updated test for HierarchyEnsembleForecaster to include conditions mentioned in the bug.

##### For all contributions
- [X] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

### CODE REVIEW

**Review by fkiraly — APPROVED** [review]
Thanks!

Ouch, this is one of those hard to spot "mutating/non-mutating" bugs, isn't it?
I suppose the "time per line" ratio must have been high here...

File: sktime/forecasting/compose/_hierarchy_ensemble.py [hunk 1]
```diff
@@ -285,7 +285,7 @@ def _get_node_dict(self, z):
                 if counter == 0:
                     nodes = mi
                 else:
-                    nodes.append(mi)
+                    nodes = nodes.append(mi)
```
Comment (yarnabrina) [inline sktime/forecasting/compose/_hierarchy_ensemble.py:288]: Is this an intended change? Do you want to change `nodes` to `None` here?

```pycon
>>>
>>> a = []
>>> a = a.append(2)
>>> a
>>> a is None
True
>>>
```

Comment (VyomkeshVyas) [reply sktime/forecasting/compose/_hierarchy_ensemble.py:288]: Thanks for reviewing. Its an intended change as nodes is not of type list rather pd.Index. I too realized after spotting the bug that append function works little different with pd.Index object.
```python
>>>
>>> a = pd.Index([1,2])
>>> b = pd.Index([3])
>>> a = a.append(b)
>>> a
Index([1, 2, 3], dtype='int64')
>>>
```

**Comment by VyomkeshVyas** [comment]
> Ouch, this is one of those hard to spot "mutating/non-mutating" bugs, isn't it? I suppose the "time per line" ratio must have been high here...
Absolutely! took me some time to find the issue.

**Comment by fkiraly** [comment]
> Absolutely! took me some time to find the issue.

Reminds me of the good old days when I learnt that `[]` is not a good parameter default...

File: sktime/forecasting/compose/_hierarchy_ensemble.py (hunk 1 above)
Comment (yarnabrina) [reply sktime/forecasting/compose/_hierarchy_ensemble.py:288]: Thanks for clarifying. In that case, can you please add a comment in the code for next time someone comes at this line? This is not intuitive at all (not to me at least), especially after seeing `nodes = []` in L278.

**Review by fkiraly — APPROVED** [review]
Still approve

**Review by yarnabrina — APPROVED** [review]

**Pull request merged**